### PR TITLE
Updates for Python 3 compatibility 

### PR DESCRIPTION
**Pull request by Adam-WL**
Useful script - Splitwise was shared with me today, needed to import a bunch of exported transactions from a CSV, but this wasn't working with my Python 3 environment.

- Updated some stuff from Python 2 to work with Python 3.10
- Couldn't figure out the stdout>oauth_verifier from the pipe so switched it to a user input step
- Added support for categorization of expenses
- Added reference table for category ID values

### CODE REVIEW

**Comment by esecules** [comment]
Thanks for the pr, I will review it soon.

File: src/groupsplit.py [hunk 1]
```diff
@@ -326,14 +345,34 @@ def main():
...
+    splitwise = Splitwise()
+    cats = splitwise.get_categories()
+    # Extract ID and Name from the list of dictionaries
+    result = []
+    for item in cats:
+        result.append({'id': item['id'], 'name': item['name']})
+        if 'subcategories' in item:
+            for subcategory in item['subcategories']:
+                result.append({'id': subcategory['id'], 'name': subcategory['name']})
+    file_name = '../test/all_categories_list.csv'
```
Comment (esecules) [inline src/groupsplit.py]: Files downloaded and used during runtime should go into a temporary folder ignored by git. Do categories in SW change per user or are they static? If it's the latter there's no need for this function.

**Review by esecules — CHANGES_REQUESTED** [review]
Please split this into 2 PRs, one to update to Python3 and the other to add this categories feature. I'll be able to merge the version update PR quickly. I also want to make some changes to the CI pipeline in the meantime.

File: src/groupsplit.py (hunk 1 above)
Comment (Adam-WL) [reply src/groupsplit.py]: I believe it's static so I'll pull this out

**Comment by Adam-WL** [comment]
Thanks for feedback - makes sense. This PR has been reduced to just the Python 3 compatibility updates.

**Comment by esecules** [comment]
I'll merge this and then add the updates i want to make. Thanks for bringing the project up to date with python3!

**Pull request merged**